game/src/game/player/mod.rs

```diff
@@ -1482,20 +1539,16 @@ pub fn on_stealth_hit_cooldown_reset(
 
 /// Exits player Stealthing state when a stealthed attack first hits
 pub fn on_hit_exit_stealthing(
-    query: Query<&Attack, (Added<Hit>, With<Stealthed>)>,
+    query: Query<&Attack, (With<Hit>, With<Stealthed>)>,
```

Comment:
currently this *should* be triggered any time hit is added to the attack, did this fix the issue you were having?